src/main/java/seedu/algobase/logic/parser/FindPlanCommandParser.java

```diff
@@ -1,35 +1,95 @@
  */
-public class FindPlanCommandParser {
+public class FindPlanCommandParser implements Parser {
+
+    private List<String> getArgumentValueAsList(String argValue) {
+        String trimmedArg = argValue.trim();
+        String[] keywords = trimmedArg.split("\\s+");
+        return Arrays.asList(keywords);
+    }
 
     /**
-     * Parses the given {@code String} of arguments in the context of the FindCommand
-     * and returns a FindCommand object for execution.
+     * Parses the given {@code String} of arguments in the context of the FindPlanCommand
+     * and returns a FindPlanCommand object for execution.
+     *
      * @throws ParseException if the user input does not conform the expected format
      */
+    @Override
     public FindPlanCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
-        if (trimmedArgs.isEmpty()) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindPlanCommand.MESSAGE_USAGE));
+        ArgumentMultimap argumentMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_DESCRIPTION,
+                        PREFIX_START_DATE, PREFIX_END_DATE);
+
+        // According to the command format, no preamble should be present.
+        if (!argumentMultimap.getPreamble().isBlank()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindPlanCommand.MESSAGE_USAGE));
         }
 
-        String[] nameKeywords = trimmedArgs.split("\\s+");
+        FindPlanDescriptor findPlanDescriptor = new FindPlanDescriptor();
 
-        return new FindPlanCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
-    }
+        if (argumentMultimap.getValue(PREFIX_NAME).isPresent()) {
+            List<String> planNameKeywords = getArgumentValueAsList(argumentMultimap.getValue(PREFIX_NAME).get());
+            List<Keyword> keywords = planNameKeywords.stream().map(Keyword::new).collect(Collectors.toList());
+            findPlanDescriptor.setPlanNamePredicate(new PlanNameContainsKeywordsPredicate(keywords));
+        }
 
+        if (argumentMultimap.getValue(PREFIX_DESCRIPTION).isPresent()) {
+            List<String> planDescriptionKeywords =
+                    getArgumentValueAsList(argumentMultimap.getValue(PREFIX_DESCRIPTION).get());
+            List<Keyword> keywords = planDescriptionKeywords.stream().map(Keyword::new).collect(Collectors.toList());
+            findPlanDescriptor.setPlanDescriptionPredicate(
+                    new PlanDescriptionContainsKeywordsPredicate(keywords));
+        }
+
+        if (argumentMultimap.getValue(PREFIX_START_DATE).isPresent()
+                || argumentMultimap.getValue(PREFIX_END_DATE).isPresent()) {
+            if (argumentMultimap.getValue(PREFIX_START_DATE).isEmpty()
+                    || argumentMultimap.getValue(PREFIX_END_DATE).isEmpty()) {
```

Comment:
Should use `isBlank`

Reply:
The logic can be confusing here. This can be handled in `parseDate`.

Reply:
@tiuweehan `argumentMultimap.getValue(PREFIX_START_DATE)` returns a `Optional`, so `isBlank` does not work here.

Reply:
'parseDate' method can only check one string at one time, so I'll still keep this.